### DISCUSSION TITLE
Restore "Last-Modified" time during apk generation.

### DISCUSSION THREAD

**Discussion by yankarinRG** [iBotPeaches/Apktool#3156] [Ideas]
### Information
1. **Apktool Version (`apktool -version`)** - 2.4.0
2. **Operating System (Mac, Linux, Windows)** - Windows 10 64-bit
3. **APK From? (Playstore, ROM, Other)** - APKMirror

### Steps to Reproduce
1. `apktool d <app>.apk`
2. Edit a file. For example, AndroidManifest.xml
3. `apktool b <app> -o <new_app>.apk`
4. Now open both <app>.apk and <new_app>.apk with 7-Zip FIle Manager or similar and compare both edited files, in our example, AndroidManifest.xml:
the original file, in <app>.apk will have the bare minimum needed metadata and last edited date of 01/01/2009, while the edited file, in <new_app>.apk will have more metadata (thus more overhead) than the original file.

**N.B.** this is just an example, all files inside the apk (zip), regardless if edited or not, will have those added metadata, **at least** the current date instead of 01/01/2009.

Performing a `zipalign` does not solve the issue.

[Zip file format FIle headers](https://en.wikipedia.org/wiki/Zip_(file_format)#File_headers)

### APK
If this APK can be freely shared, please upload/attach a link to it.
[Google Phone](https://www.apkmirror.com/apk/google-inc/google-phone/google-phone-37-0-260231449-publicbeta-release/google-phone-37-0-260231449-publicbeta-2-android-apk-download/)

### Questions to ask before submission
1. Have you tried `apktool d`, `apktool b` without changing anything? Yes
2. If you are trying to install a modified apk, did you resign it? Yes
3. Are you using the latest apktool version? Yes

### Attached pictures:
In **red**, you can notice that by removing from the apk (zip) the file android-support-multidex.version and putting it back again, metadata is added.
In **green** , date difference between original and edited app.

![app](https://user-images.githubusercontent.com/23347445/62141806-7adfee80-b2ed-11e9-8312-64342c8636b3.jpg)
![app_new](https://user-images.githubusercontent.com/23347445/62141808-7adfee80-b2ed-11e9-8a25-a36bf6f5c6b2.jpg)

**Comment by iBotPeaches** [comment]
So is there a bug you are encountering with this behavior?

**Comment by yankarinRG** [comment]
Not that I know or have discovered, and I think there aren't. It's just that it would be cleaner because the apk file would be more similar to a genuine apk.

**Comment by iBotPeaches** [comment]
Okay, I did some investigation and we could probably do this by recording the LastModified time of files during read, then re-set them back to that value after writing zip file.

However, system is not really designed to do that at all. So marking as feature request for backlog.

**Comment by mimi89999** [comment]
Hello,
I think that this feature is very important for reproducible rebuilds. Could you please make it possible to set that date to an arbitrary value?
